File: AppValueCalculator/app/src/main/java/com/example/appvaluecalculator/MainActivity.java
package com.example.appvaluecalculator;

import static java.lang.Integer.parseInt;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText nilai_tugas, nilai_uts, nilai_uas;
    Button hitung, keluar;
    TextView hasil_tugas, hasil_uts, hasil_uas, nilai_akhir;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        nilai_tugas = (EditText) findViewById(R.id.input_ntugas);
        nilai_uts = (EditText) findViewById(R.id.input_nuts);
        nilai_uas = (EditText) findViewById(R.id.input_nuas);

        hitung = (Button) findViewById(R.id.button_hitung);
        keluar = (Button) findViewById(R.id.button_keluar);

//        double ntugas = Double.parseDouble(nilai_tugas.getText().toString()) * 30 / 100;
//        hasil_tugas.setText(Double.toString(ntugas));
//        double nuts = Double.parseDouble(nilai_uts.getText().toString()) * 35 / 100;
//        hasil_uts.setText(Double.toString(nuts));
//        double nuas = Double.parseDouble(nilai_uas.getText().toString()) * 35 * 100;
//        hasil_uas.setText(Double.toString(nuas));
//
//        hitung.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                if((nilai_tugas.getText().length()>0) && (nilai_uts.getText().length()>0) && (nilai_uas.getText().length()>0)){
//                    double result = ntugas + nuts + nuas;
//                    nilai_akhir.setText(Double.toString(result));
//                }
//            }
//        });

    }
}

File: BasicApplicationKotlin/app/src/main/java/com/example/basicapplicationkotlin/MainActivity.kt
package com.example.basicapplicationkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    // Kode Lateinit
    private lateinit var inputEditText: EditText
    private lateinit var sendButton: Button
    private lateinit var helloTextView: TextView

    private fun initComponents(){
        inputEditText = findViewById(R.id.inputEditText)
        sendButton = findViewById(R.id.sendButton)
        helloTextView = findViewById(R.id.helloTextView)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initComponents()

        helloTextView.text = "Hi"

        sendButton.setOnClickListener {
            val name = inputEditText.text.toString()
            helloTextView.text = "Hi, $name"
        }
    }
}